languagetool-language-modules/sv/src/main/java/org/languagetool/rules/sv/UpperCaseNgramRule.java

```diff
@@ -0,0 +1,103 @@
+/* LanguageTool, a natural language style checker 
+ * Copyright (C) 2020 Daniel Naber (http://www.danielnaber.de)
+ * 
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+package org.languagetool.rules.sv;
+
+import org.languagetool.AnalyzedSentence;
+import org.languagetool.AnalyzedTokenReadings;
+import org.languagetool.Language;
+import org.languagetool.languagemodel.LanguageModel;
+import org.languagetool.UserConfig;
+import org.languagetool.rules.*;
+import org.languagetool.rules.ngrams.Probability;
+
+import java.util.*;
+
+import static org.languagetool.tools.StringTools.*;
+
+/**
+ * Finds some(!) words written uppercase that should be spelled lowercase and vice versa.
+ * @since 6.2+
+ */
+public class UpperCaseNgramRule extends Rule {
+
+  private static final int THRESHOLD = 50;
+  private static final Set<String> relevantWords = new HashSet<>(Arrays.asList(
+    "maj", "Maj", 
+    "Måndag", "måndag"
+  ));
+
+  private final LanguageModel lm;
+
+  public UpperCaseNgramRule(ResourceBundle messages, LanguageModel lm, Language langUser, UserConfig userConfig) {
+    super(messages);
+    super.setCategory(Categories.CASING.getCategory(messages));
+    this.lm = Objects.requireNonNull(lm);
+    //setDefaultTempOff();  // fixme!
+    setLocQualityIssueType(ITSIssueType.Misspelling);
+    addExamplePair(Example.wrong("Antagningen sker 15 <marker>Maj</marker>."),
+                   Example.fixed("Antagningen sker 15 <marker>maj</marker>."));
+  }
+
+  @Override
+  public final String getId() {
+    return "SV_UPPER_CASE_NGRAM";
+  }
+
+  @Override
+  public String getDescription() {
+    return "Granskar ord som ofta skrivs med liten eller stor begynnelsebokstav när det ska vara tvärsom";
+  }
+
+  @Override
+  public RuleMatch[] match(AnalyzedSentence sentence) {
```

Comment:
This is a copy of `UpperCaseNgramRule` for German, isn't it? We should avoid that kind of code duplication.

Reply:
Yes, I have looked at what would be the best foundation for each of the features and started from there. But I have definitely not just copied it. It is all carefully selected to make initial testing possible for each feature. But if you have a suggestion on how to add the ngram feature without duplicating some extending methods which happens to be extended in a few other languages as well?

Reply:
These methods seem to be identical except the `String msg` part, aren't they? Then all the identical parts should be in a super class in `languagetool-core` and only the messages should be defined in the sub classes.

Reply:
Yes, as an initial dummy for creating the tests. Ok, I will start with removing it and then add it according to your suggestion. There are way fewer implementations, three only, to keep track of here. :)